presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -404,7 +404,7 @@ private Type handleResolvedField(Expression node, FieldId fieldId, Field field,
                 }
             }
 
-            if (field.getOriginTable().isPresent() && field.getName().isPresent()) {
+            if (field.getOriginTable().isPresent() && field.getName().isPresent() && !field.isAliased()) {
```

Comment:
If we track this and expose it as part of the query result metadata, this would allow us to implement JDBC [ResultSetMetaData](https://docs.oracle.com/javase/10/docs/api/java/sql/ResultSetMetaData.html) methods like `getColumnName` and `getTableName`.

Reply:
I didn't realize isAlias wasn't for any reference to a user-defined column name.  thanks!  I updated the PR to track origin column names.

@electrum  I didn't expose it anywhere, but it will be available now to be exposed if someone is working on JDBC ResultSetMetadata.